samza-core/src/main/java/org/apache/samza/execution/JobGraph.java

```diff
@@ -248,7 +248,7 @@ private StreamEdge getOrCreateStreamEdge(StreamSpec streamSpec, boolean isInterm
     String streamId = streamSpec.getId();
     StreamEdge edge = edges.get(streamId);
     if (edge == null) {
-      boolean isBroadcast = appDesc.getBroadcastStreams().contains(streamId);
+      boolean isBroadcast = appDesc.getIntermediateBroadcastStreamIds().contains(streamId);
```

Comment:
Is it possible that the streamSpec is for a non-intermediate broadcast stream? Then it seems like this check would return false, but it should be true.

Reply:
> Is it possible that the streamSpec is for a non-intermediate broadcast stream
We currently don't support specifying broadcast inputs using input descriptors. 
SAMZA-1841 will address that later. Currently these stream IDs are those that were generated using the High Level API broadcast() operator, hence the renaming to clarify that this is only for intermediate streams.

Does that answer your question?

Reply:
Got it, so appDesc.getBroadcastStreams was already not including broadcast inputs. That makes sense that your change isn't changing anything, but does that mean there is currently an issue with getOrCreateStreamEdge which doesn't consider broadcast inputs through config?